classes/EventRegistry.php

```diff
@@ -263,4 +265,67 @@ public function pruneTranslatedContentTemplates($templates)
             return !Str::endsWith($template->getBaseFileName(), $extensions);
         });
     }
+
+    /**
+     * Adds language suffixes to mail view files.
+     * @param  \October\Rain\Mail\Mailer $mailer
+     * @param  \Illuminate\Mail\Message $message
+     * @param  string $view
+     * @param  array $data
+     * @param  string $raw
+     * @param  string $plain
+     * @return string|null
+     */
+    public function findLocalizedMailViewContent($mailer, $message, $view, $data, $raw, $plain)
+    {
+        if (isset($raw) || (!isset($view) && !isset($plain))) {
+            return;
+        }
+
+        $locale = strtolower(isset($data['_current_locale']) ? $data['_current_locale'] : App::getLocale());
+
+        $factory = $mailer->getViewFactory();
+
+        if (isset($view)) {
+            $view = $this->getLocalizedView($factory, $view, $locale);
+        }
+
+        if (isset($plain)) {
+            $plain = $this->getLocalizedView($factory, $plain, $locale);
+        }
+
+        if (($code = $view ?: $plain) === null) {
+            return;
+        }
+
+        return !MailManager::instance()->addContentToMailer($message, $code, $data, $view === null);
```

Comment:
What parameter is being based by $view === null? And why is the return result negated?

Reply:
$plainOnly
```
modules/system/classes/MailManager.php:    protected function addContentToMailerInternal($message, $template, $data, $plainOnly = false)
```
the return is negated because when `addContentToMailer()` returns true, it means the content has been successfully added and we don't want the function firing the event continue its execution (see below)

src/Mail/Mailer.php:
```
        if (
            ($this->fireEvent('mailer.beforeAddContent', [$message, $view, $data, $raw, $plain], true) === false) ||
            (Event::fire('mailer.beforeAddContent', [$this, $message, $view, $data, $raw, $plain], true) === false)
        ) { 
            return;
        }
```

Reply:
Why return anything then if we don't want the execution to stop after the event handler is processed? addContentToMailer can return false, which means that this handler would stop execution if it failed to do its job?

Reply:
if addContentToMailer() returns true, then it sent the mail using the localized view.

if addContentToMailer() returns false, then we WANT the regular flow to continue to send the mail using the original "non-localized" view.

Reply:
If that's the case then why is there the `!` in front of it?

Reply:
I just explained it above @LukeTowers . See my last commit, it makes it more readable.